Add 2.10.18 Release Notes (#10111)
PrefectHQ/prefect: RELEASE-NOTES.md

- Add support for deployment triggers to `prefect deploy` — https://github.com/PrefectHQ/prefect/pull/10049
- Add `prefect.yaml` `pull` step generation based on `build_docker_image` `build` step  — https://github.com/PrefectHQ/prefect/pull/10090

### Fixes
- Fix pull step saving by preserving placeholders with missing values — https://github.com/PrefectHQ/prefect/pull/10053
- Fix windows file limit check — https://github.com/PrefectHQ/prefect/pull/10059
- Allows use of Prefect variables in `job_variables` section of deploy config in `prefect.yaml` — https://github.com/PrefectHQ/prefect/pull/10078
- Add default option to new_parameters.pop in `explode_variadic_parameter` used to handle `**kwargs` in task mapping — https://github.com/PrefectHQ/prefect/pull/10067
- Update Docker-based `prefect init` recipes to use `push_docker_image` step — https://github.com/PrefectHQ/prefect/pull/10092
- Update `prefect deploy` to skip schedule prompts if `schedule` key present in `prefect.yaml` — https://github.com/PrefectHQ/prefect/pull/10074
- Fix saving of `pull` and `push` step deployment configuration — https://github.com/PrefectHQ/prefect/pull/10087

> 

Reviewer: ```suggestion
- Fix issue hosting and running the UI in unsecured contexts - https://github.com/PrefectHQ/prefect-design/pull/829

```

# Prefect Release Notes

## Release 2.10.18

### Docker image support during flow deployment

> We have introduced enhanced support for Docker-based infrastructures when deploying flows through the interactive `prefect deploy` experience. Users can now easily custom-build or auto-build Docker images and push them to remote registries if they so choose.

Reviewer: ```suggestion
We enhanced support for Docker-based infrastructures when deploying flows through the interactive `prefect deploy` experience. Users can now easily custom-build or auto-build Docker images and push them to remote registries if they so choose.
```

> The CLI automatically detects if a work pool supports Docker images (e.g., docker, ecs, cloud-run) during `prefect deploy`, guiding the user through the experience of building and pushing a Docker image.

Reviewer: ```suggestion
The CLI automatically detects if a work pool supports Docker images (e.g., docker, ecs, cloud-run) during `prefect deploy` and will now guide the user through the experience of building and pushing a Docker image if support is detected.
```

This enhancement to managing deployments will greatly simplify the process of creating the necessary `build` and `push` steps for deployments.

Not only that, we will also create a `pull` step for you when you choose to build a Docker image through `prefect deploy`. Whether you have your own Dockerfile or you want to use the auto-build feature in `build_docker_image`, we will create a `pull` step for you to help you set the correct path to your flow code.

> See the following pull request for implementation details:

Reviewer 2: ```suggestion
See the following pull requests for implementation details:
```

- https://github.com/PrefectHQ/prefect/pull/10022
- https://github.com/PrefectHQ/prefect/pull/10090

### Event-driven deployments with triggers

> You can now add event-based triggers to your deployments in Prefect Cloud, allowing you to add triggers into your `prefect.yaml` file or from the Prefect UI deployment page. Deployment triggers use automations under the hood, and any automations that trigger a deployment will be linked to from the deployments page. 

Reviewer: ```suggestion
You can now easily incorporate event-based triggers into your Prefect Cloud deployments - simply add triggers to your `prefect.yaml` file or directly from the Prefect UI deployment page. Deployment triggers utilize automations - any automation that runs flows from a given deployment will be reflected on that deployment page.
```
I cleaned this up a little for clarity/brevity, feel free to reject this

> See the following pull request for implementation details:

Reviewer 2: ```suggestion

See the following pull requests for implementation details:
```

# Prefect Release Notes

## Release 2.10.18

### Event driven deployments with triggers
Deployments in Prefect Cloud now have an optional triggers field, which describes what events (or lack thereof) should trigger a flow run. You can add triggers to deployments in your .yaml files or in the UI from your deployment page. Deployment triggers use automations under the hood, and any automations that trigger a deployment will be linked to from the deployments page. 

> 

Reviewer 2: ```suggestion
See the following pull request for implementation details:
- https://github.com/PrefectHQ/prefect/pull/10049
```

The CLI automatically detects if a work pool supports Docker images (e.g., docker, ecs, cloud-run) during `prefect deploy`, guiding the user through the experience of building and pushing a Docker image.

This enhancement to managing deployments will greatly simplify the process of creating the necessary `build` and `push` steps for deployments.

Not only that, we will also create a `pull` step for you when you choose to build a Docker image through `prefect deploy`. Whether you have your own Dockerfile or you want to use the auto-build feature in `build_docker_image`, we will create a `pull` step for you to help you set the correct path to your flow code.

See the following pull request for implementation details:
- https://github.com/PrefectHQ/prefect/pull/10022
- https://github.com/PrefectHQ/prefect/pull/10090

### Enhancements
- Allow saving of updated deployment configurations — https://github.com/PrefectHQ/prefect/pull/10018

> - Add support for deployment triggers to `prefect deploy` — https://github.com/PrefectHQ/prefect/pull/10049

Reviewer 2: Moved to highlight
```suggestion
```

> - Add `prefect.yaml` `pull` step generation based on `build_docker_image` `build` step  — https://github.com/PrefectHQ/prefect/pull/10090

Reviewer 2: Referenced in highlight
```suggestion
```

> - Webhook creation UI (cloud only)

Reviewer 2: ?

### Fixes

> - Fix pull step saving by preserving placeholders with missing values — https://github.com/PrefectHQ/prefect/pull/10053

Reviewer 2: ```suggestion
- Fix deployment `pull` step saving by preserving placeholders with missing values — https://github.com/PrefectHQ/prefect/pull/10053
```

- Fix windows file limit check — https://github.com/PrefectHQ/prefect/pull/10059
- Allows use of Prefect variables in `job_variables` section of deploy config in `prefect.yaml` — https://github.com/PrefectHQ/prefect/pull/10078

> - Add default option to new_parameters.pop in `explode_variadic_parameter` used to handle `**kwargs` in task mapping — https://github.com/PrefectHQ/prefect/pull/10067

Reviewer 2: ```suggestion
- Add default option to `new_parameters.pop` in `explode_variadic_parameter` used to handle `**kwargs` in task mapping — https://github.com/PrefectHQ/prefect/pull/10067
```

> - Update Docker-based `prefect init` recipes to use `push_docker_image` step — https://github.com/PrefectHQ/prefect/pull/10092

Reviewer 2: This is also not a fix

- Update `prefect deploy` to skip schedule prompts if `schedule` key present in `prefect.yaml` — https://github.com/PrefectHQ/prefect/pull/10074
- Fix saving of `pull` and `push` step deployment configuration — https://github.com/PrefectHQ/prefect/pull/10087

### Documentation
- Adjust docs to reflect Prefect requires Python 3.8 — https://github.com/PrefectHQ/prefect/pull/9853

> - Add custom pull step examples with variables — https://github.com/PrefectHQ/prefect/pull/10073

Reviewer 2: ```suggestion
- Add custom `pull` step examples to deployment management docs — https://github.com/PrefectHQ/prefect/pull/10073
```

# Prefect Release Notes

## Release 2.10.18

> ### Event driven deployments with triggers

Reviewer 2: ```suggestion
### Event-driven deployments with triggers
```
I like the dash :-)

> Deployments in Prefect Cloud now have an optional triggers field, which describes what events (or lack thereof) should trigger a flow run. You can add triggers to deployments in your .yaml files or in the UI from your deployment page. Deployment triggers use automations under the hood, and any automations that trigger a deployment will be linked to from the deployments page. 

Reviewer 2: I actually don't feel we should begin with a Prefect Cloud highlight. I think we should move this down.

- Fix windows file limit check — https://github.com/PrefectHQ/prefect/pull/10059
- Allows use of Prefect variables in `job_variables` section of deploy config in `prefect.yaml` — https://github.com/PrefectHQ/prefect/pull/10078
- Add default option to new_parameters.pop in `explode_variadic_parameter` used to handle `**kwargs` in task mapping — https://github.com/PrefectHQ/prefect/pull/10067
- Update Docker-based `prefect init` recipes to use `push_docker_image` step — https://github.com/PrefectHQ/prefect/pull/10092
- Update `prefect deploy` to skip schedule prompts if `schedule` key present in `prefect.yaml` — https://github.com/PrefectHQ/prefect/pull/10074
- Fix saving of `pull` and `push` step deployment configuration — https://github.com/PrefectHQ/prefect/pull/10087

### Documentation
- Adjust docs to reflect Prefect requires Python 3.8 — https://github.com/PrefectHQ/prefect/pull/9853
- Add custom pull step examples with variables — https://github.com/PrefectHQ/prefect/pull/10073
- Add troubleshooting guide to docs — https://github.com/PrefectHQ/prefect/pull/10079

> - Add example for finding Prefect Cloud Account ID and Workspace ID — https://github.com/PrefectHQ/prefect/pull/10103

Reviewer 2: it wasn't a true example, so I changed the wording

- Add links to Webhooks documentation from Events documentation — https://github.com/PrefectHQ/prefect/pull/10045

> - Simply deployment description — https://github.com/PrefectHQ/prefect/pull/10050

Reviewer 2: ```suggestion
- Simplify deployment description in docs — https://github.com/PrefectHQ/prefect/pull/10050
```

> - Add trigger to deployment fields — https://github.com/PrefectHQ/prefect/pull/10097

Reviewer 2: Related to deployment trigger highlight, should be moved up there

> - Add trigger to deployment fields — https://github.com/PrefectHQ/prefect/pull/10097

Reviewer 2: ```suggestion
```

Related to deployment trigger highlight, should go up there

### Contributors

> - @masonmenges made their first contribution in https://github.com/PrefectHQ/prefect/pull/9977
> - @garylavayou made their first contribution in https://github.com/PrefectHQ/prefect/pull/10060
> - @themattmorris made their first contribution in https://github.com/PrefectHQ/prefect/pull/10056
> - @aimeemcmanus made their first contribution in https://github.com/PrefectHQ/prefect/pull/10076

Reviewer 2: ```suggestion
- @garylavayou made their first contribution in https://github.com/PrefectHQ/prefect/pull/10060
- @themattmorris made their first contribution in https://github.com/PrefectHQ/prefect/pull/10056
```
excluding Prefect employees, despite their contributions being very much appreciated

> - @NodeJSmith
> - @garylavayou
> - @rpeden
> - @themattmorris

Reviewer 2: ```suggestion
- @NodeJSmith
- @rpeden
```
aforementioned
